### PR TITLE
i#3544 RV64: Enable the build of the drcov test for RV64

### PR DESCRIPTION
**Pull request by chenhy0106**
1. Enable the document generation for drcov2lcov options as RISC-V binaries now work.
2. Enable the build of the drcov test linux.eintr for RISC-V. This test can be built and runs correctly on the physical machine. However, due to qemu issues, it cannot be added to the RUNS_ON_QEMU list for RISC-V now.

Issue: #3544

### CODE REVIEW

**Review by ksco — DISMISSED** [review]

**Review by ksco — COMMENTED** [review]
Wait, does these newly opened tests checked by CI? We need to add them to RUNS_IN_QEMU for that, right?

File: clients/drcov/CMakeLists.txt [hunk 1]
```diff
@@ -175,20 +175,15 @@ get_property(dox_targets GLOBAL PROPERTY DynamoRIO_dox_targets)
 set_property(GLOBAL PROPERTY DynamoRIO_dox_targets ${dox_targets} ${doctgt})
```
Comment (derekbruening) [inline clients/drcov/CMakeLists.txt:175]: > i#3544 RV64: Open drcov test for RV64

nit: I find "Open" confusing: not sure what it means here.  "Enable" seems much clearer.

**Review by derekbruening — APPROVED** [review]

**Comment by chenhy0106** [comment]
> Wait, does these newly opened tests checked by CI? We need to add them to RUNS_IN_QEMU for that, right?

Only build the `eintr` test without adding it to `RUNS_IN_QEMU`. Due to some issues with QEMU, the tests cannot pass under QEMU. Now, tests that involve creating threads with `clone` can not pass under qemu-riscv64.

Testing on the physical machine works fine.

File: clients/drcov/CMakeLists.txt (hunk 1 above)
Comment (chenhy0106) [reply clients/drcov/CMakeLists.txt:175]: Ok.

File: suite/tests/CMakeLists.txt [hunk 2]
```diff
@@ -3657,7 +3657,7 @@ if (BUILD_CLIENTS)
     set(tool.drcov.fib_expectbase "tool.drcov.fib")
     DynamoRIO_get_full_path(tool.drcov.fib_postcmd drcov2lcov "${location_suffix}")
 
-    if (UNIX AND NOT RISCV64) # TODO i#3544: Port tests to RISC-V 64
+    if (UNIX)
       # Test an app that executes a pipe syscall for i#5981.
       torunonly_ci(tool.drcov.eintr linux.eintr drcov linux/eintr.c "" "" "")
```
Comment (derekbruening) [inline suite/tests/CMakeLists.txt:3662]: Add a `TODO #3544` comment about adding this to the RUNS_ON_QEMU list, explaining it's a QEMU issue that's preventing it.  Maybe better to put that comment down by the RUNS_ON_QEMU list itself.  Let's not have future readers think you made a mistake and forgot to add it when you enabled the build: use comments to make an omission like this clear.

Comment (derekbruening) [inline suite/tests/CMakeLists.txt:3662]: Update the PR description (will become commit message) to also say that this test is not added to RUNS_ON_QEMU yet.

**Review by derekbruening — APPROVED** [review]

File: suite/tests/CMakeLists.txt (hunk 2 above)
Comment (chenhy0106) [reply suite/tests/CMakeLists.txt:3662]: https://github.com/DynamoRIO/dynamorio/blob/a2f8bcc95def89de92141e627e53c796b1377089/suite/tests/CMakeLists.txt#L6300
I see. But there's already such a comment before the RUNS_ON_QEMU list.

Comment (chenhy0106) [reply suite/tests/CMakeLists.txt:3662]: OK. I will update the PR description.

Comment (derekbruening) [reply suite/tests/CMakeLists.txt:3662]: That doesn't imply that those tests succeed by themselves on bare metal: it just says they build.

**Pull request merged**